### ISSUE TITLE
Add explicit copy constructor comment

### ISSUE THREAD

**Issue by jbendes** [ZeroCM/zcm#49]
We should add a comment to the emitted zcmtypes that indicate that they intentionally have no copy constructor explicitly defined; they use the implicit copy constructor. This works because there is no dynamic memory in zcmtypes. Any dynamic fields are represented as vectors which have copy constructors defined in stdlib.

**Comment by olsoni** [comment]
It would actually be best to simply add `= default` declarations for all copy, move, and assignment operators because this would make it obvious to users that they can make use of the implicit ones.

**Comment by jbendes** [comment]
Does that work for non c++11 compilers?

**Comment by olsoni** [comment]
No, but if that's an issue, we should add a --use-c++11 option because it's silly not to leverage that functionality.

**Comment by jbendes** [comment]
But then we're forcing people to have c++11 capable compilers if they want
to use zcmtypes in c++. Seems like a strange requirement to enforce. Maybe
theres a #ifdef we could use to include those default declarations only if
the user is compiling with the 11 standard?

**Comment by olsoni** [comment]
not sure how to respond to this other than quoting my previous comment:

> if that's an issue, we should add a --use-c++11 option because it's silly not to leverage that functionality

ie add a ./waf configure option that enables the c++11 features like the forced default of move constructors. Note that copy constructors can be `=default` in pre-c++11, it's just that move constructors weren't explicitly introduced until c++11

**Comment by jbendes** [comment]
I pushed up a branch showing what I meant with my suggestion. Let me know what you think.
[c63917e](https://github.com/ZeroCM/zcm/commit/c63917e4a945ad810565d26430c18704089585fd)

**Comment by olsoni** [comment]
Close, but because the "c++11" check requires it to actually be compiled with -std=c++11, not just linked to code using c++11, we need to add a waf configure option `--use-c++11` that will change the `-std` flag in the c++ compiliation

**Comment by jbendes** [comment]
I dont think I understand your last comment here @olsoni. Why can we not merge this?
